pkg/doc/jose/jwk_test.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jose
+
+import (
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rand"
+	"testing"
+
+	"github.com/decred/dcrd/dcrec/secp256k1/v2"
+	"github.com/square/go-jose/v3/json"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHeaders_GetJWK(t *testing.T) {
+	headers := Headers{}
+
+	pubKey, _, err := ed25519.GenerateKey(rand.Reader)
+	require.NoError(t, err)
+
+	jwk := JWK{
+		Key:       pubKey,
+		KeyID:     "kid",
+		Algorithm: "EdDSA",
+	}
+
+	jwkBytes, err := json.Marshal(jwk)
+	require.NoError(t, err)
+
+	var jwkMap map[string]interface{}
+
+	err = json.Unmarshal(jwkBytes, &jwkMap)
+	require.NoError(t, err)
+
+	headers["jwk"] = jwkMap
+
+	parsedJWK, ok := headers.JWK()
+	require.True(t, ok)
+	require.NotNil(t, parsedJWK)
+
+	// jwk is not present
+	delete(headers, "jwk")
+	parsedJWK, ok = headers.JWK()
+	require.False(t, ok)
+	require.Nil(t, parsedJWK)
+
+	// jwk is not a map
+	headers["jwk"] = "not a map"
+	parsedJWK, ok = headers.JWK()
+	require.False(t, ok)
+	require.Nil(t, parsedJWK)
+}
+
+func TestDecodePublicKey(t *testing.T) {
+	t.Run("Test decode public key success", func(t *testing.T) {
+		tests := []struct {
+			name    string
+			jwkJSON string
+		}{
+			{
+				name: "get public key bytes Ed25519 JWK",
+				jwkJSON: `{
+							"kty": "OKP",
+							"use": "enc",
+							"crv": "Ed25519",
+							"kid": "sample@sample.id",
+							"x": "sEHL6KXs8bUz9Ss2qSWWjhhRMHVjrog0lzFENM132R8",
+							"alg": "EdDSA"
+						}`,
+			},
+			{
+				name: "get public key bytes RSA JWK",
+				jwkJSON: `{
+							"kty": "RSA",
+							"e": "AQAB",
+							"use": "enc",
+							"kid": "sample@sample.id",
+							"alg": "RS256",
+							"n": "1hOl09BUnwY7jFBqoZKa4XDmIuc0YFb4y_5ThiHhLRW68aNG5Vo23n3ugND2GK3PsguZqJ_HrWCGVuVlKTmFg` +
+					`JWQD9ZnVcYqScgHpQRhxMBi86PIvXR01D_PWXZZjvTRakpvQxUT5bVBdWnaBHQoxDBt0YIVi5a7x-gXB1aDlts4RTMpfS9BPmEjX` +
+					`4lciozwS6Ow_wTO3C2YGa_Our0ptIxr-x_3sMbPCN8Fe_iaBDezeDAm39xCNjFa1E735ipXA4eUW_6SzFJ5-bM2UKba2WE6xUaEa5G1` +
+					`MDDHCG5LKKd6Mhy7SSAzPOR2FTKYj89ch2asCPlbjHTu8jS6Iy8"
+						}`,
+			},
+			{
+				name: "get public key bytes EC P-526 JWK",
+				jwkJSON: `{
+							"kty": "EC",
+							"use": "enc",
+							"crv": "P-256",
+							"kid": "sample@sample.id",
+							"x": "JR7nhI47w7bxrNkp7Xt1nbmozNn-RB2Q-PWi7KHT8J0",
+							"y": "iXmKtH0caOgB1vV0CQwinwK999qdDvrssKhdbiAz9OI",
+							"alg": "ES256"
+						}`,
+			},
+			{
+				name: "get public key bytes EC SECP256K1 JWK",
+				jwkJSON: `{
+							"kty": "EC",
+    						"use": "enc",
+    						"crv": "secp256k1",
+    						"kid": "sample@sample.id",
+    						"x": "wQehEGTVCu32yp8IwTaBCqPUIYslyd-WoFRsfDKE9II",
+    						"y": "rIJO8RmkExUecJ5i15L9OC7rl7pwmYFR8QQgdM1ERWI",
+   						 	"alg": "EdDSA"
```

Comment:
This doesn't look right: `"alg": "EdDSA"`
with
`"crv": "secp256k1",`

secp256k1 is an ECDSA curve to my knowledge. We only support Ed25519 as EdDSA curve.

Reply:
that's only sample input I generated for tests, you are right EdDSA also works.